tt_metal/tt_metal.cpp

```diff
@@ -976,7 +976,7 @@ uint32_t CreateSemaphore(
 
 std::shared_ptr<Buffer> CreateBuffer(const InterleavedBufferConfig &config) {
     return std::make_shared<Buffer>(
-        config.device, config.size, config.page_size, config.buffer_type, config.buffer_layout);
+        config.device, config.size, config.page_size, config.buffer_type, config.buffer_layout, std::nullopt, std::nullopt, config.allocate);
```

Comment:
sorry for the late review, generally looks okay to me. I don't love that we have to pass in the nullopt to the buffer constructor. It would probably be better for the buffer ctor to accept buffer configs. This isn't in the scope of this PR so Ill open up a separate issue for that :)